### PR TITLE
Changes to when_floating rules

### PR DESCRIPTION
**Pull request by karthink**
when_floating is None by default
when set to True, apply rule to floating windows only
when set to False, apply rule to tiled windows only

As per discussion in #2846.

I haven't been able to test this code. When I start Qtile with startx with the path to this directory, my Xserver crashes. Is there an easier way to test Qtile?

### CODE REVIEW

**Comment by m-col** [comment]
Awesome, thanks!

>  Is there an easier way to test Qtile?

In the source directory, check out `scripts/xephyr`. It uses `Xephyr` to run a nested X session in a window within your main X session. For this you will need `Xephyr` installed. Check out http://docs.qtile.org/en/latest/manual/hacking.html#using-xephyr, and dependencies are also listed at the top of that page.

**Comment by m-col** [comment]
Are there any docs that need updating to reflect this?

Also it might be a good idea to add an entry to CHANGELOG about this so others can find the feature!

**Comment by m-col** [comment]
Looks like your branch has conflicts with master so you'll need to rebase. If needed, there is some information on how to do this at: http://docs.qtile.org/en/latest/manual/howto/git.html#i-was-told-to-rebase-my-work

It might be good to also squash your commits down into a single commit while you're there - there's also a section on that page for that in case you find it useful.

**Comment by karthink** [comment]
I added a line in the example code illustrating the difference.

**Comment by m-col** [comment]
The changes look good but the CI is complaining about inconsistent indentation in the changelog entry. Make sure that the indentation of your entry matches the rest of the file.

**Comment by elParaguayo** [comment]
Can we fix the indentation in changelog, resolve the conflict and rebase into a single commit so we can get this merged?

**Comment by karthink** [comment]
I fixed the indentation and rebased on the latest master.

**Comment by elParaguayo** [comment]
Can you squash everything into a single commit (see http://docs.qtile.org/en/latest/manual/howto/git.html#i-was-told-to-squash-some-commits).

**Comment by karthink** [comment]
OK, I think I managed to rebase on master and squash commits. Hope it looks right now.

File: libqtile/lazy.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def __init__(self, call: CommandGraphCall, args: Tuple, kwargs: Dict) -> None:
         self._kwargs = kwargs
 
         self._layouts: Set[str] = set()
-        self._when_floating = True
+        self._when_floating = None
```
Comment (elParaguayo) [inline libqtile/lazy.py:50]:
```suggestion
        self._when_floating: Optional[bool] = None
```
This is the initial declaration of `when_floating` so you need to declare the type here.

This should fix the mypy error you're seeing.

**Review by elParaguayo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**